src/main/java/org/digma/intellij/plugin/toolwindow/recentactivity/RecentActivityService.java

```diff
@@ -1,47 +1,389 @@
+import static org.digma.intellij.plugin.toolwindow.common.ToolWindowUtil.parseJsonToObject;
+import static org.digma.intellij.plugin.ui.list.insights.JaegerUtilKt.openJaegerFromRecentActivity;
+import static org.digma.intellij.plugin.ui.list.insights.JaegerUtilKt.traceButtonName;
 
 public class RecentActivityService implements Disposable {
 
-    private final Logger logger = Logger.getInstance(RecentActivityService.class);
+    private static final String DIGMA_SIDE_PANE_TOOL_WINDOW_NAME = "Digma";
+    private static final String RESOURCE_FOLDER_NAME = "recentactivity";
+    private static final String RECENT_EXPIRATION_LIMIT_VARIABLE = "recentActivityExpirationLimit";
+    private static final int FETCHING_LOOP_INTERVAL = 10 * 1000; // 10sec
 
+    private final Logger logger = Logger.getInstance(RecentActivityService.class);
+    private final Icon icon = AppIcons.TOOL_WINDOW_OBSERVABILITY;
+    private final Icon iconWithGreenDot = ExecutionUtil.getLiveIndicator(icon);
+    private final String localHostname;
     private final Project project;
-    private JBCefBrowser jbCefBrowser;
+    private final AnalyticsService analyticsService;
 
     //the recent activity code is not managed in one place that is accessible from the plugin code
     // like a project service, so currently need an init task.
     // it is used to send live data in case the live view button was clicked before the recent activity window was initialized.
-    private MyInitTask initTask = null;
+    private MyInitTask initTask;
+    private Timer myLiveDataTimer;
+    private Timer activityFetchingTimer;
+    private JBCefBrowser jbCefBrowser;
+    private RecentActivityResult latestActivityResult;
+    private boolean webAppInitialized;
 
-    private Timer myLiveDataTimer = null;
+    public static RecentActivityService getInstance(Project project) {
+        return project.getService(RecentActivityService.class);
+    }
 
     public RecentActivityService(Project project) {
         this.project = project;
+        //initialize AnalyticsService early so the UI already can detect the connection status when created
+        this.analyticsService = project.getService(AnalyticsService.class);
+        this.localHostname = CommonUtils.getLocalHostname();
+        this.latestActivityResult = new RecentActivityResult(null, new ArrayList<>());
     }
 
+    public void startFetchingActivities() {
+        if (activityFetchingTimer != null)
+            return;
 
-    public static RecentActivityService getInstance(Project project){
-        return project.getService(RecentActivityService.class);
+        var activityFetchingTask = new TimerTask() {
+            @Override
+            public void run() {
+                fetchRecentActivities();
+            }
+        };
+        activityFetchingTimer = new Timer();
+        activityFetchingTimer.scheduleAtFixedRate(activityFetchingTask, 0, FETCHING_LOOP_INTERVAL);
+    }
```

Comment:
I think that fixed-delay execution is more appropriate here then fixed-rate execution. like myLiveDataTimer

from the java docs:
In fixed-rate execution, each execution is scheduled relative to the scheduled execution time of the initial execution. If an execution is delayed for any reason (such as garbage collection or other background activity), two or more executions will occur in rapid succession to "catch up."

In fixed-delay execution, each execution is scheduled relative to the actual execution time of the previous execution. If an execution is delayed for any reason (such as garbage collection or other background activity), subsequent executions will be delayed as well

Reply:
fixed